dockertest.go

```diff
@@ -545,7 +545,23 @@ func (d *Pool) Retry(op func() error) error {
 	bo := backoff.NewExponentialBackOff()
 	bo.MaxInterval = time.Second * 5
 	bo.MaxElapsedTime = d.MaxWait
-	return backoff.Retry(op, bo)
```

Comment:
The documentation (https://pkg.go.dev/github.com/cenkalti/backoff/v4#Retry) indicates that:

> Retry the operation o until it does not return error or BackOff stops. o is guaranteed to be run at least once.
>
>If o returns a *PermanentError, the operation is not retried, and the wrapped error is returned.
>
>Retry sleeps the goroutine for the duration returned by BackOff after a failed operation returns.

Could you maybe explain why using the channel here is better?